### PR TITLE
STT Draft

### PR DESCRIPTION
**Pull request by PaulBlogger13**
Not sure what i am doing

### CODE REVIEW

File: STTBot/events/app_mention.py [hunk 1]
```diff
@@ -394,6 +402,16 @@ def _ret_success_blocks(blocks, suc_message, say):
...
         "func": _cmd_poll_react
     },
+    {
+        "cmd": "STT",
+        "sub_cmd": "Draft",
+        "args": [
+            "list"
+        ],
+        "help": "Randomises order of users for STT Draft",
+        "func": _cmd_STT_draft
```
Comment (krdme) [inline STTBot/events/app_mention.py:413]: This needs to match the function name on 305

File: STTBot/events/app_mention.py [hunk 2]
```diff
@@ -394,6 +402,16 @@ def _ret_success_blocks(blocks, suc_message, say):
         "help": "Adds reactions for pin showdown",
         "func": _cmd_poll_react
     },
+    {
+        "cmd": "STT",
+        "sub_cmd": "Draft",
+        "args": [
+            "list"
```
Comment (krdme) [inline STTBot/events/app_mention.py]: `list_of_users` might be a better variable name x

File: STTBot/events/app_mention.py [hunk 3]
```diff
@@ -302,7 +302,15 @@ def _cmd_pin_remove(client, event_data, command, say):
     data_interface.remove_pin(permalink.channel, permalink.timestamp)
     return {"message": ":white_check_mark: Successfully removed pin"}
 
-
+def _STT_draft(client, event_data, command, say):
+    if len(command.args) == 0;
+        raise CommandError("No list provided")
+    else:
+        draft_order = command.args
+        random.shuffle(draft_order)
```
Comment (krdme) [inline STTBot/events/app_mention.py:311]: You need to import random at the top of the file for this to work

File: STTBot/events/app_mention.py [hunk 4]
```diff
@@ -302,7 +302,15 @@ def _cmd_pin_remove(client, event_data, command, say):
...
+        
+    return {"message": ":robot_face: Order generated:", draft_order}
```
Comment (krdme) [inline STTBot/events/app_mention.py]: This is invalid dictionary syntax, the draft order should be part of the second string, i.e.

`{"message": f":robot_face: Order generated: {draft_order}"}`

However, draft_order is a list so it won't embed into the string particularly nicely by default. Look into the `join` string function to see how to convert a list to a string.

File: STTBot/events/app_mention.py (hunk 1 above)
Comment (PaulBlogger13) [reply STTBot/events/app_mention.py:413]: done, silly error

File: STTBot/events/app_mention.py (hunk 3 above)
Comment (PaulBlogger13) [reply STTBot/events/app_mention.py:311]: done

File: STTBot/events/app_mention.py (hunk 2 above)
Comment (PaulBlogger13) [reply STTBot/events/app_mention.py]: agree

File: STTBot/events/app_mention.py (hunk 4 above)
Comment (PaulBlogger13) [reply STTBot/events/app_mention.py]: hopefully now resolved

**Pull request closed without merging**